torch/distributions/transforms.py

```diff
@@ -824,7 +824,7 @@ def _inverse(self, y):
         y_cumsum_vec = tril_matrix_to_vec(y_cumsum_shifted, diag=-1)
         t = y_vec / (y_cumsum_vec).sqrt()
         # inverse of tanh
-        x = ((1 + t) / (1 - t)).log() / 2
+        x = (t.log1p() - t.neg().log1p()) / 2
```

Comment:
I suspect the `t / (1 - t)` will lose us some precision, but I guess it's a good balance between performance and precision.

Reply:
Correct, we've intentionally avoided log-of-ratio computations because they are numerically unstable.  We've preferred `log(x) - (-x).log1p()` or `.logit()` as numerically stable alternatives.

Reply:
fair enough and good to know!